File: homework/Thu_Pham/lesson_1/136-single-number.js
//Solution for https://leetcode.com/problems/single-number/

/**
 * @param {number[]} nums
 * @return {number}
 */
 var singleNumber = function(nums) {
    let res = nums[0]
    for (let i = 1; i < nums.length; i++) {
        res ^= nums[i]
    }
    return res
};

//runtime O(n) with n is the length of the input array
//space O(1) extra space

File: homework/Manh_Nguyen_Van/lesson-1/136-single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        if(nums.length==1){
            return nums[0];
        }
        int singleNumber = nums[0];
        for(int i=1; i<nums.length; i++){
            singleNumber=singleNumber^nums[i];
        }
        return singleNumber;
    }
}